### PR TITLE
SPV: Support for VertexIndex and InstanceIndex

### PR DESCRIPTION
**Pull request by greg-lunarg**

### CODE REVIEW

File: Core/metadata.h [hunk 1]
```diff
@@ -202,6 +202,8 @@ enum EMdInputOutput {
     EMioPipeIn,             // normal user-streamed input data: attributes, varyings, etc.
     EMioVertexId,
     EMioInstanceId,
+    EMioVertexIndex,
```
Comment (johnkslang) [inline Core/metadata.h]: I don't think this should be added here.

Can't this be entirely handled by adding the built-ins?  Note that list is much larger than this list, and generally, EMioPipeIn and EMioPipeOut are metacategories of a large number of built-ins.  This is the new trend, that the built-ins were added as subcategories of things in this list, and the only reason something might be in both lists is totally for backward compatibility reasons, but that wouldn't apply to something new.

(That is, it certainly doesn't make sense to have them in both lists.)

See comments in glslang's regarding TStorageQualifier and TBuiltInVariable for more context.

If correct, we should document EMioPipeIn/out as super categories, as in glslang.

**Comment by johnkslang** [comment]
It looks like there is a smaller simpler change that would also make most the diffs disappear; see code comments.

**Comment by greg-lunarg** [comment]
OK. I think it is ready for another look.

File: Frontends/SPIRV/SpvToTop.cpp [hunk 2]
```diff
@@ -911,7 +913,9 @@ gla::EMdInputOutput SpvToTopTranslator::getMdQualifier(spv::Id resultId) const
     case gla::EmbPointSize:   mdQualifier = gla::EMioPointSize;      break;
     case gla::EmbClipVertex:  mdQualifier = gla::EMioClipVertex;     break;
     case gla::EmbInstanceId:  mdQualifier = gla::EMioInstanceId;     break;
+    case gla::EmbInstanceIndex: mdQualifier = gla::EMioPipeIn;         break;
```
Comment (johnkslang) [inline Frontends/SPIRV/SpvToTop.cpp]: Why doesn't the default catch this?  Note that almost all built-in variables are missing in this list, so curious why the new ones show up as special cases.

**Comment by greg-lunarg** [comment]
Try again.

**Pull request merged**